### PR TITLE
Cleanup annotateAst

### LINKED ISSUE #17: Make annotateAst recursive
# Description

If we encounter an AST node which isn't supported while annotating, we don't descend down that branch. Due to this problem, some nodes which we can annotate don't get annotated.

# Sub-tasks

- [x] Make sure all supported AST nodes get annotated

### PR DESCRIPTION
**Pull request by HectorPeeters**
# Description

With this merge request, all unknown nodes get traversed as well. This makes sure all supported nodes in the whole AST get annotated. It also cleans up annotateAst.ts a bit.

Closes #17 

# Changes

1. remove annotateUpperLevel
2. add 'generic' annotation for unknown nodes

# Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

# How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

File: src/typeInference/annotateAST.ts [hunk 1]
```diff
@@ -77,6 +60,9 @@ export function annotate(ast: IAST): SequenceType | undefined {
 			]);
 			return { type: ValueType.XSSTRING, mult: SequenceMultiplicity.EXACTLY_ONE };
 		default:
+			for (let i = 1; i < ast.length; i++) {
+				annotate(ast[i] as IAST);
```
Comment (yuanzexiong) [inline src/typeInference/annotateAST.ts:64]: Can we make sure that it won't be stuck in a recursion loop?

Comment (HectorPeeters) [reply src/typeInference/annotateAST.ts:64]: The whole test suit passes and the parser will never generate circular AST trees so this shouldn't be a problem.

Comment (yuanzexiong) [reply src/typeInference/annotateAST.ts:64]: Okay cool well in that case it should be fine

**Review by FBroy — APPROVED** [review]

**Review by MTimmerije — APPROVED** [review]
Seems good, improves code quality and seems perfectly suitable.

**Pull request merged**